src/gl_depth_sim/mesh_loader.cpp

```diff
@@ -11,36 +11,40 @@ static std::unique_ptr<gl_depth_sim::Mesh> process(const aiScene* scene)
   const int num_meshes = scene->mNumMeshes;
   std::cout << "Scene has " << num_meshes << " meshes\n";
 
-  if (num_meshes != 1)
-  {
-    std::cerr << "Scene must have 1 mesh";
-    return {};
-  }
 
   gl_depth_sim::EigenAlignedVec<Eigen::Vector3f> vertices;
   std::vector<unsigned> indices;
 
-  const aiMesh* mesh = scene->mMeshes[0];
 
-  const int nface = mesh->mNumFaces;
-  const int nvert = mesh->mNumVertices;
-  std::cout << "Mesh has n faces/verts: " << nface << "/" << nvert << "\n";
-
-  for (int i = 0; i < nvert; ++i)
+  for(int meshNum = 0 ; meshNum <num_meshes ; meshNum++)
   {
-    const aiVector3D& v =  mesh->mVertices[i];
-    vertices.push_back({v.x, v.y, v.z});
-  }
+    const aiMesh* mesh = scene->mMeshes[meshNum];
 
-  for (int i = 0; i < nface; ++i)
-  {
-    const aiFace& f = mesh->mFaces[i];
-    assert(f.mNumIndices == 3);
-    indices.push_back(f.mIndices[0]);
-    indices.push_back(f.mIndices[1]);
-    indices.push_back(f.mIndices[2]);
-  }
+    const int nface = mesh->mNumFaces;
+    const int nvert = mesh->mNumVertices;
+    std::cout << "Mesh has n faces/verts: " << nface << "/" << nvert << "\n";
+
+    for (int i = 0; i < nvert; ++i)
+    {
+      const aiVector3D& v =  mesh->mVertices[i];
+      vertices.push_back({v.x, v.y, v.z});
+    }
+
+    int skippedFaces = 0;
+    for (int i = 0; i < nface; ++i)
+    {
+      const aiFace& f = mesh->mFaces[i];
 
+      if( f.mNumIndices == 3 ){
```

Comment:
For our code, please follow the style guide for ROS. There's a clang format plugin in the ROS QTC editor I believe that can handle most of this for you.